index.js

```diff
@@ -111,61 +99,101 @@ Rollout.prototype.get = function (key, id, opt_values, multi) {
-    keys.push(key + ':' + k, percentage_map[k])
+Rollout.prototype.update = function (key, modifierPercentages) {
+  var persistKeys = [], modName, p
+  for (modName in modifierPercentages) {
+    p = normalizePercentageRange(modifierPercentages[modName])
+    persistKeys.push(key + ':' + modName, JSON.stringify(p))
   }
-  return new Promise(function (resolve, reject) {
-    this.client.mset(keys, function (err, result) {
-      if (err) return reject(err)
-      resolve(result)
-    })
-  }.bind(this))
+  return setRedisKeys(this.client, persistKeys)
 }
 
-Rollout.prototype.mods = function (name) {
+Rollout.prototype.modifiers = function (handlerName) {
   var keys = []
-  var names = []
-  for (var flag in this._handlers[name]) {
-    keys.push(name + ':' + flag)
-    names.push(flag)
+  var modNames = []
+  for (var modName in this._handlers[handlerName]) {
+    keys.push(handlerName + ':' + modName)
+    modNames.push(modName)
   }
-  return new Promise(function (resolve, reject) {
-    this.client.mget(keys, function (err, result) {
-      if (err) return reject(err)
-      resolve(result)
-    })
-  }.bind(this))
-    .then(function (values) {
-      var flags = {}
-      values.forEach(function (val, i) {
-        flags[names[i]] = val
-      })
-      return flags
+  return getRedisKeys(this.client, keys)
+  .then(function (values) {
+    var modifiers = {}
+    values.forEach(function (val, i) {
+      modifiers[modNames[i]] = JSON.parse(val)
     })
+    return modifiers
+  })
 }
 
-Rollout.prototype.flags = function () {
-  return Object.keys(this._handlers)
+Rollout.prototype.handlers = function () {
+  return Promise.resolve(Object.keys(this._handlers))
 }
 
 Rollout.prototype.val_to_percent = function (text) {
   var n = crypto.createHash('md5').update(text).digest('hex')
   n = n.slice(0, n.length/2)
   return parseInt(n, 16) / parseInt(n.split('').map(function () { return 'f' }).join(''), 16) * 100
 }
+
+function defaultCondition() {
+  return true
+}
+
+function clampPercentage(val) {
+  return Math.max(0, Math.min(100, +(val || 0)))
+}
+
+function normalizePercentageRange(val) {
+  if (val && typeof val === 'object') {
+    return {
+      min: clampPercentage(val.min),
+      max: clampPercentage(val.max)
+    }
+  }
+  return clampPercentage(val)
+}
+
+function isPercentageInRange(val, range) {
+  // Redis stringifies everything, so ranges must be reified
+  if (typeof range === 'string') {
+    range = JSON.parse(range)
+  }
+  if (range && typeof range === 'object') {
+    return val > range.min && val <= range.max
+  }
+  return val < range
+}
+
+function getRedisKeys(client, keys) {
+  return new Promise(function (resolve, reject) {
+    client.mget(keys, promiseCallback(resolve, reject))
+  })
+}
+
+function setRedisKeys(client, keys) {
+  return new Promise(function (resolve, reject) {
+    client.mset(keys, promiseCallback(resolve, reject))
+  })
+}
+
+function promiseCallback(resolve, reject) {
```

Comment:
Note for followup - we should promisify the functions instead of using these callbacks